### ISSUE TITLE
RTD in CI does not respect GitHub approval requirements

### ISSUE THREAD

**Issue by pllim** [readthedocs/readthedocs.org#11000]
## What's the problem this feature will solve?

When a first-time contributor submits a PR to OSS, GitHub requires approval to run the CI to prevent abuse (for better or worse, see https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks). When I approved it, I noticed that RTD job was already completed for https://docs.readthedocs.io/en/stable/pull-requests.html , which means RTD job ran regardless of this requirement.

## Describe the solution you'd like

I think it might be desirable to have RTD job tied to a PR also follows the GitHub procedures, so that if GitHub thinks CI needs approval to run, RTD should wait for the approval to come in first.

## Alternative solutions

If you cannot tie in to GitHub approval machinery, maybe roll your own, or at least document that you do not follow their approval rules?

## Additional context

Not everyone might agree with this request. If you disagrees, feel free to close this issue without further actions. Thanks.

**Comment by webknjaz** [comment]
I don't think that GH exposes any CI approval APIs or that there's an easy way to re-implementing the same.

**Comment by humitos** [comment]
We are currently using GitHub OAuth application, which is the "old way" to integrate with GitHub. I doubt they are exposing an API for OAuth application and implementing this may require us to migrate to a GitHub Application (the "new way"). However, when we explored that path, it was not trivial and complex without too much value added.

We may need to review the situation now and re-consider the migration. After that, we could probably jump into the approval requirements. In the meantime, I'd appreciate any link to documentation that GitHub has about how to do this.
